python/opencc/__init__.py

```diff
@@ -47,15 +47,15 @@
 class OpenCC(object):
 
     def __init__(self, config='t2s.json'):
+        if not config.endswith('.json'):
```

Comment:
same here

Reply:
If the C++ API is unable to find a file to source the dictionary from, python would throw segfault. This isn't particularly informative and is not inherently a bug.
This here is to ensure that resource specified by the user exists.

Reply:
If the config does not exist, the C library should anyway fail.

I don't know why we have to try to add '.json' to the config.

Reply:
Let me explain using an example

```python
>>> from opencc import libopencc
>>> converter = libopencc.opencc_open('opencc/clib/share/opencc/jp2s'.encode())
>>> libopencc.opencc_convert_utf8(converter, 'word'.encode(), 4)
[1]    6586 segmentation fault (core dumped)
```
`jp2s` is not a valid config and as expected, the C library fails so python throws segmentation fault. This is an expected behavior and not a bug.

But a python developer might not interpret as such since debug messages are suppressed.

I've configured the python API to fail if a reference file cannot be found. Effectively, this is akin to anticipating c errors by first throwing a python error.

```python
>>> import opencc
>>> opencc.convert('word', 'jp2s')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/OpenCC/python/dist/opencc/__init__.py", line 89, in convert
    cc = OpenCC(config)
  File "/opt/OpenCC/python/dist/opencc/__init__.py", line 67, in __init__
    raise ValueError('Could not find file at {}'.format(config))
ValueError: Could not find file at /opt/OpenCC/python/dist/opencc/clib/share/opencc/jp2s.json
```
This error message is much more meaningful.

Reply:
Why is there a segmentation fault? opencc_open should return an error if the config does not exist.

Reply:
`¯\_(ツ)_/¯` I can only guess that somewhere in the [cpython](https://github.com/python/cpython) source code, they coded it such that seg-fault happens when a c library error is thrown.

Reply:
Ok. I am fine. But we should fix this properly in future.